src/filerepository.js

```diff
@@ -174,10 +176,20 @@ export default class FileRepository extends Plugin {
 			return null;
 		}
 
-		const loader = new FileLoader( file );
+		const filePromise = fileOrPromise instanceof Promise ? fileOrPromise : new Promise( resolve => resolve( fileOrPromise ) );
+		const loader = new FileLoader( filePromise );
+
 		loader._adapter = this.createUploadAdapter( loader );
```

Comment:
I've just stumbled upon this and didn't realized it earlier. The docs for `FileLoader`'s constructor are rubbish then.

It's a chicken and egg problem as Loader needs adapter and creating adapter requires loader :D

Please remove it from constructor's docs and maybe create some privete method to hide this obscurity :)

I'd leave here:
```js
const loader = this._initLoader( fileOrPromise );
```
so all the above code would go the the `initLoader` method. Also please add some coment above `loader._adaper = ` part there as it should be explained why this trick ( AFAICS the `this.createUploadAdapter()` expects the loader instance.

Reply:
Didn't notice it too. 😂 🐔 Maybe instead of doing `loader._adapter` which looks bad, we could pass `createUploadAdapter` function as a second param of a `FileLoader` constructor:

```js
class FileLoader {
	constructor( filePromise, createAdapterFn ) {
        this._adapter = createAdapterFn( this );
    }
}
```

it makes it a little more complicated and coupled, but as `FileRepository` creates adapter only to "pass" it to the loader, maybe loader should create adapter itself? I'm not sure though, WDYT?